src/components/activityEventElems/ActivityElement/ActivityElement.tsx

```diff
@@ -87,27 +124,177 @@ function Subject({ subject }: { subject: string | JSX.Element | null }) {
   )
 }
 
-export function ActivityEvent({
+function _ActivityEvent({
   header,
   subject,
-  event,
+  details,
   extra,
+  style,
 }: {
-  header: string
+  header: string | JSX.Element | null
   subject: string | JSX.Element | null
   extra?: string | JSX.Element | null
-  event: ActivityElementEvent
+  details: Partial<ActivityElementEvent>
+  style?: CSSProperties
 }) {
   return (
-    <>
+    <div style={style}>
       <DetailsContainer>
         <div>
           <Header header={header} />
           <Subject subject={subject} />
         </div>
-        <SideDetails event={event} />
+        <SideDetails details={details} />
       </DetailsContainer>
       {extra ? <ExtraContainer>{extra}</ExtraContainer> : null}
-    </>
+    </div>
+  )
+}
+
+export function ActivityEvent({
+  event,
+  style,
+}: {
+  event: ProjectEvent
+  style?: CSSProperties
+}) {
+  // TODO: we should use v2 where we can, or a better context
+  const { tokenSymbol } = useContext(V1ProjectContext)
+
+  const { Header, Subject, Extra, details } = useMemo(() => {
+    if (event.payEvent) {
+      return {
+        Header: PayEventHeader,
+        Subject: <PayEventSubject amount={event.payEvent.amount} />,
+        Extra: (
+          <PayEventExtra
+            feeFromV2Project={event.payEvent.feeFromV2Project ?? undefined}
+            note={event.payEvent.note}
+          />
+        ),
+        details: { ...event.payEvent, type: 'beneficiary' as const },
+      }
+    }
+    if (event.redeemEvent) {
+      return {
+        Header: RedeemEventHeader,
+        Subject: (
+          <RedeemEventSubject
+            amount={event.redeemEvent.amount}
+            tokenSymbol={tokenSymbol}
+          />
+        ),
+        Extra: (
+          <RedeemEventExtra returnAmount={event.redeemEvent.returnAmount} />
+        ),
+        details: { ...event.redeemEvent, type: 'beneficiary' as const },
+      }
+    }
+    if (event.projectCreateEvent) {
+      return {
+        Header: ProjectCreateHeader,
+        Subject: (
+          <ProjectCreateSubject caller={event.projectCreateEvent.caller} />
+        ),
+        Extra: null,
+        details: event.projectCreateEvent,
+      }
+    }
+    if (event.deployedERC20Event) {
+      return {
+        Header: DeployedERC20EventHeader,
+        Subject: (
+          <DeployedERC20EventSubject symbol={event.deployedERC20Event.symbol} />
+        ),
+        extra: null,
```

Comment:
should be `Extra` not `extra` eh? maybe we should type the return of this useMemo